common/headtracker/types/heads.go

```diff
@@ -0,0 +1,18 @@
+package types
+
+import (
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// Heads is a collection of heads. All methods are thread-safe.
+type Heads[H types.Head[BLOCK_HASH], BLOCK_HASH types.Hashable] interface {
+	// LatestHead returns the block header with the highest number that has been seen, or nil.
+	LatestHead() H
```

Comment:
Do these method names need to have "Head"? Seems like they could be simplified

Reply:
I think that the verbosity could help especially if we might have different function for `Latest`, for instance `LatestFinalizedHead` suggested by @prashantkumar1982 .

Latest could also mean Latest `Heads` instead of `Head`

Reply:
Ya I don't have strong opinions.
Mostly just go with what is the common-practice in golang on our codebase.
Whatever we choose, stick to that type for consistency.

Reply:
The Go idiom to follow here is to not repeat yourself, which is why I asked. If there aren't method name collisions from combining with other interfaces, then repeating 'head(s)' seems redundant.